clang/lib/Driver/Driver.cpp

```diff
@@ -6369,6 +6369,10 @@ const char *Driver::GetNamedOutputPath(Compilation &C, const JobAction &JA,
   if (AtTopLevel && !isa<DsymutilJobAction>(JA) && !isa<VerifyJobAction>(JA)) {
     if (Arg *FinalOutput = C.getArgs().getLastArg(options::OPT_o))
       return C.addResultFile(FinalOutput->getValue(), &JA);
+    // Output to destination for -fsycl-device-only and Windows -o
+    if (C.getArgs().hasArg(options::OPT_fsycl_device_only))
```

Comment:
What is it about the MSVC variant of the option that requires this additional condition check?

Reply:
There are certain conditions when using -o on Windows that expects to add file extensions (like when performing `clang-cl -o file -c test.cpp`) which will create `file.obj` instead of just `file`.

Reply:
Ah! So we're basically talking about enabling "non-typical" formats for named device-only outputs (like `.out` instead of `.bc`/`.ll`)??

Reply:
The purpose is to enable `-o` behavior for Windows which matches `-o` behavior for Linux targets.  Right now, using `-o file` doesn't even output to an expected file (uses default file name)